api/src/main/java/io/grpc/Metadata.java

```diff
@@ -733,6 +736,15 @@ private static BitSet generateValidTChars() {
     private static String validateName(String n, boolean pseudo) {
       checkNotNull(n, "name");
       checkArgument(!n.isEmpty(), "token must have at least 1 tchar");
+      if (n.equals("connection")) {
```

Comment:
Is this always lower case?

Reply:
Yep. https://github.com/grpc/grpc-java/blob/7c6f53ab79846decde5e624bb6fa5f7a1f8a922c/api/src/main/java/io/grpc/Metadata.java#L750